pywbemtools/pywbemcli/_common.py

```diff
@@ -892,52 +896,86 @@ def create_params(classname, cim_method, kv_params):
 
 def sort_cimobjects(cim_objects):
     """
-    Sort lists of CIMClass, CIMCLassName, CIMQualifierDecl, CIMInstance or
-    CIMInstanceName. Sorts based on name or CIMInstancename. Sorting is based
-    on the name value (name, classname, wbemuri(canonical form)
-    Returns new list with the sorted objects.  This was defined as a common
-    sort mechanism for all of the CIM object responses from WBEM servers.
+    Return a sorted list of the input objects.
+
+    The returned list is always a new list and its items are the original input
+    objects.
+
+    The following input object types are supported, and their sort key is:
+
+    * string: Sorted by string value (case sensitively). This case covers the
+      result of 'class enumerate'.
+
+    * CIMClass: Sorted by class name (case sensitively).
+
+    * CIMClassName, CIMInstanceName: Sorted by its canonical WBEM URI string.
+      This makes the sort case insensitive.
+
+    * CIMInstance: Must have the path set. Sorted by the canonical WBEM URI
+      string of its instance path. This makes the sort case insensitive.
 
-    For classes it sorts by classnames. For instances and instancenames, it
-    sorts by instancename. For qualifier declarations it sorts by name
+    * CIMQualifierDeclaration: Sorted by qualifier name (case sensitively).
+
+    * tuple(CIMClassName, CIMClass): Sorted by the canonical WBEM URI
+      string of the CIMClassName object. This makes the sort case insensitive.
+      This case covers the result of 'class references/associators'.
 
     Parameters:
 
-      cim_objects (list of CIMObjects):
+      cim_objects (Sequence): Objects to be sorted.
 
     Returns:
-        A list of the same types as input but sorted as defined above.
+        A new list of the original input objects, sorted as defined above.
+
+    Raises:
+        TypeError: Invalid type for being sorted.
```

Comment:
Change to Invalid type for input parameter cim_object.

Reply:
DONE